code/modules/events/koi_mirgration.dm

```diff
@@ -27,12 +21,12 @@
 	while(i <= num_groups)
 		var/group_size = rand(group_size_min, group_size_max)
 		for(var/j = 1, j <= group_size, j++)
-			var/mob/living/simple_animal/hostile/retaliate/carp/koi/K = new(spawn_locations[i])
-			spawned_koi.Add(K)
+			var/carptype = pickweight(spawned_mobs)
+			spawned_carp.Add(new carptype(spawn_locations[i]))
 		i++
 
-/datum/event/koi_migration/end()
-	for(var/mob/living/simple_animal/hostile/retaliate/carp/koi/C in spawned_koi)
+/datum/event/carp_migration/koi/end()
```

Comment:
You can axe this proc entirely - make sure to make the `/mob/living/simple_animal/hostile/retaliate/carp/C` loop in the parent version into something more like `/mob/living/L` or something